### PR TITLE
feat: adds search and pagination to gcp access view

### PR DESCRIPTION
**Pull request by kneeraj2000**

### CODE REVIEW

File: gcp/urls.py [hunk 1]
```diff
@@ -1,8 +1,7 @@
 from django.urls import re_path
 
-from .views import get_gcp_domains, get_gcp_groups
+from .views import get_gcp_groups
 
 urlpatterns = [
-    re_path(r"^api/v1/gcp/domains$", get_gcp_domains),
     re_path(r"^api/v1/gcp/domain/groups", get_gcp_groups),
```
Comment (niveshm) [inline gcp/urls.py:6]: change from re_path to path

File: gcp/views.py [hunk 2]
```diff
@@ -15,14 +12,28 @@ def get_gcp_groups(request):
         data["gcp_domain"]
```
Comment (niveshm) [inline gcp/views.py:12]:
```suggestion
        data.get("gcp_domain")
```
change this

File: gcp/views.py [hunk 3]
```diff
@@ -15,14 +12,28 @@ def get_gcp_groups(request):
...
     ):
         return JsonResponse({"error": "Valid domain is required for GCP Access."})
```
Comment (niveshm) [inline gcp/views.py:14]: add status code
```suggestion
        return JsonResponse({"error": "Valid domain is required for GCP Access."}, status=400)
```

File: gcp/views.py [hunk 4]
```diff
@@ -15,14 +12,28 @@ def get_gcp_groups(request):
...
+    gcp_groups_data = access.GCPAccess.get_domain_groups(gcp_domain)
+
+    groups = []
+    all_groups = []
+    for gcp_group in gcp_groups_data:
+        if search.lower() in gcp_group["name"].lower():
+            groups.append(gcp_group["name"])
+        all_groups.append(gcp_group["name"])
+    
+    paginator = Paginator(groups, 10) if groups else Paginator(all_groups, 10)
```
Comment (niveshm) [inline gcp/views.py:26]: use the paginator decorator from Access.paginator_decorators file

**Review by vedharish — APPROVED** [review]

**Pull request merged**